pull-kueue-test-e2e-kueueviz-main fail with webhook errors

<!--
Please use this template for reporting flaky tests.
Links to specific failures in Prow are appreciated.
-->

**Which test is flaking?**:
pull-kueue-test-e2e-kueueviz-main

**Link to failed CI job or steps to reproduce locally**:
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/12003/pull-kueue-test-e2e-kueueviz-main/2064060201723498496

**Failure message or logs**:
```
workloadpriorityclass.kueue.x-k8s.io/long-term-research created
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/00-resource-flavor.yaml": Internal error occurred: failed calling webhook "mresourceflavor.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/00-resource-flavor.yaml": Internal error occurred: failed calling webhook "mresourceflavor.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/00-resource-flavor.yaml": Internal error occurred: failed calling webhook "mresourceflavor.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-resourceflavor?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01-cluster-queues.yaml": Internal error occurred: failed calling webhook "mclusterqueue.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01-cluster-queues.yaml": Internal error occurred: failed calling webhook "mclusterqueue.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01-cluster-queues.yaml": Internal error occurred: failed calling webhook "mclusterqueue.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01-cluster-queues.yaml": Internal error occurred: failed calling webhook "mclusterqueue.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-kueue-x-k8s-io-v1beta2-clusterqueue?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01.5-cohorts.yaml": Internal error occurred: failed calling webhook "vcohort.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s](https://kueue-webhook-service.kueue-system.svc/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01.5-cohorts.yaml": Internal error occurred: failed calling webhook "vcohort.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s](https://kueue-webhook-service.kueue-system.svc/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01.5-cohorts.yaml": Internal error occurred: failed calling webhook "vcohort.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s](https://kueue-webhook-service.kueue-system.svc/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01.5-cohorts.yaml": Internal error occurred: failed calling webhook "vcohort.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s](https://kueue-webhook-service.kueue-system.svc/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/01.5-cohorts.yaml": Internal error occurred: failed calling webhook "vcohort.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s](https://kueue-webhook-service.kueue-system.svc/validate-kueue-x-k8s-io-v1beta2-cohort?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/03-agi-job.yaml": Internal error occurred: failed calling webhook "mjob.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-batch-v1-job?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-batch-v1-job?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/04-llm-job.yaml": Internal error occurred: failed calling webhook "mjob.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-batch-v1-job?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-batch-v1-job?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Error from server (InternalError): error when creating "/home/prow/go/src/sigs.k8s.io/kueue/cmd/kueueviz/examples/05-cancer-cure-research.yaml": Internal error occurred: failed calling webhook "mjob.kb.io": failed to call webhook: Post "[https://kueue-webhook-service.kueue-system.svc:443/mutate-batch-v1-job?timeout=10s](https://kueue-webhook-service.kueue-system.svc/mutate-batch-v1-job?timeout=10s)": dial tcp 10.96.15.9:443: connect: connection refused
Cleaning up kueueviz processes
/home/prow/go/src/sigs.k8s.io/kueue/hack/testing/e2e-kueueviz-backend.sh: line 37: kill: `': not a pid or valid job spec
make: *** [Makefile-test.mk:694: test-e2e-kueueviz] Error 1
```

**Anything else we need to know?**:

It looks like KueueViz e2e tests don't wait for Kueue to be available.

--- a/hack/testing/e2e-kueueviz-backend.sh
+++ b/hack/testing/e2e-kueueviz-backend.sh
@@ -34,7 +34,9 @@ fi
 # Function to clean up background processes
 cleanup() {
   echo "Cleaning up kueueviz processes"
-  kill "${BACKEND_PID}"
+  # BACKEND_PID may be unset if we exit before the backend starts (e.g. while
+  # waiting for the Kueue webhook), so guard against an empty kill argument.
+  [ -n "${BACKEND_PID:-}" ] && kill "${BACKEND_PID}"
   cluster_collect_artifacts "${KIND_CLUSTER_NAME}" ""
   cluster_cleanup "${KIND_CLUSTER_NAME}"
 }
@@ -50,7 +52,11 @@ cluster_kind_load "${KIND_CLUSTER_NAME}"
 kueue_deploy
 kubectl wait deploy/kueue-controller-manager -n"$KUEUE_NAMESPACE" --for=condition=available --timeout=5m
 
-# Deploy KueueViz resources
+# Wait for the Kueue webhook to be up before creating resources, otherwise the
+# creates can fail with "connection refused". A server-side dry-run probes the
+# webhook without persisting anything, so it is safe to retry.
+"${ROOT_DIR}/hack/testing/retry.sh" --attempts 10 --delay 3 --stream -- \
+  kubectl create --dry-run=server -f "${ROOT_DIR}/cmd/kueueviz/examples/00-resource-flavor.yaml"
 kubectl create -f "${ROOT_DIR}/cmd/kueueviz/examples/"
 
 # Start KueueViz backend

--- a/hack/testing/e2e-kueueviz-local.sh
+++ b/hack/testing/e2e-kueueviz-local.sh
@@ -26,7 +26,10 @@ mkdir -p "${ARTIFACTS}"
 # Function to clean up background processes
 cleanup() {
   echo "Cleaning up kueueviz processes"
-  kill $BACKEND_PID $FRONTEND_PID
+  # The PIDs may be unset if we exit before the processes start (e.g. while
+  # waiting for the Kueue webhook), so guard against empty kill arguments.
+  [ -n "${BACKEND_PID:-}" ] && kill "${BACKEND_PID}"
+  [ -n "${FRONTEND_PID:-}" ] && kill "${FRONTEND_PID}"
   cluster_collect_artifacts "$KIND_CLUSTER_NAME" ""
   cluster_cleanup "$KIND_CLUSTER_NAME"
 }
@@ -41,7 +44,11 @@ cluster_kind_load "${KIND_CLUSTER_NAME}"
 kueue_deploy
 kubectl wait deploy/kueue-controller-manager -n"$KUEUE_NAMESPACE" --for=condition=available --timeout=5m
 
-# Deploy kueueviz resources
+# Wait for the Kueue webhook to be up before creating resources, otherwise the
+# creates can fail with "connection refused". A server-side dry-run probes the
+# webhook without persisting anything, so it is safe to retry.
+"${ROOT_DIR}/hack/testing/retry.sh" --attempts 10 --delay 3 --stream -- \
+  kubectl create --dry-run=server -f "${ROOT_DIR}/cmd/kueueviz/examples/00-resource-flavor.yaml"
 kubectl create -f "${ROOT_DIR}/cmd/kueueviz/examples/"
 
 # Start kueueviz backend

